graph_weather/models/atmorep/model/decoder.py

```diff
@@ -0,0 +1,96 @@
+from typing import List
+
+import torch
+import torch.nn as nn
+
+
+class Decoder(nn.Module):
+    """
+    A Decoder module designed to process features of a given field,
+    using transformer blocks with U-Net-like skip connections for efficient
+    multi-resolution feature propagation.
+
+    Args:
+        hidden_dim (int): Dimensionality of the hidden embeddings.
+        num_layers (int): Number of layers (transformer blocks) in the decoder.
+        num_heads (int): Number of attention heads for each transformer block.
+        dropout (float, optional): Dropout probability applied within transformer blocks.
+            Defaults to 0.1.
+        attention_dropout (float, optional): Dropout probability for attention weights.
+            Defaults to 0.1.
+        transformer_block_cls (optional): Custom transformer block class if you want
+            to inject your own implementation. Defaults to `TransformerBlock`.
+    """
+
+    def __init__(
+        self,
+        hidden_dim: int,
+        num_layers: int,
+        num_heads: int,
+        dropout: float = 0.1,
+        attention_dropout: float = 0.1,
+        transformer_block_cls=nn.Module,  # replace with your actual block class if needed
```

Comment:
I would remove this bit or  if set to a default, default to the one that works with the decoder. But the class is used in a quite specific way, so I would recommend removing this as a configuration option.